napari/layers/labels/labels.py

```diff
@@ -340,6 +351,36 @@ def selected_label(self, selected_label):
         self._selected_color = self.get_color(selected_label)
         self.events.selected_label()
 
+    @property
+    def enable_custom_color_dict(self):
+        """True if enabling custom color dictionary"""
+        return (
+            hasattr(self, '_enable_custom_color_dict')
+            and self._enable_custom_color_dict
+        )
+
+    @enable_custom_color_dict.setter
+    def enable_custom_color_dict(self, enable_custom_color_dict):
+
+        if not self.color_dict:
+            return
+
+        if enable_custom_color_dict:
+            custom_colormap, custom_color = colormaps.color_dict_to_colormap(
+                self.color_dict
+            )
+            self.colormap = custom_colormap
+            self.custom_color = custom_color
```

Comment:
Can we rename `custom_color` to the private `_label_color_index` here and elsewhere as it is sort of analogous to how we use `_label_index`